lib/api/unittest/CAnomalyJobTest.cc

```diff
@@ -751,4 +751,106 @@ BOOST_AUTO_TEST_CASE(testRestoreFailsWithEmptyStream) {
+            "Supplied description for snapshot at " +
+            ml::core::CTimeUtils::toIso8601(expectedSnapshotTimestamp)};
+
+        std::ostringstream ostrm;
+        ostrm << expectedSnapshotTimestamp << " " << expectedSnapshotId << " "
+              << expectedSnapshotDescription;
+
+        const std::string& validPersistControlMessage{ostrm.str()};
+
+        ml::core_t::TTime snapshotTimestamp;
+        std::string snapshotId;
+        std::string snapshotDescription;
+        BOOST_TEST_REQUIRE(ml::api::CAnomalyJob::parsePersistControlMessage(
+            validPersistControlMessage, snapshotTimestamp, snapshotId, snapshotDescription));
+
+        BOOST_TEST_REQUIRE(expectedSnapshotTimestamp == snapshotTimestamp);
+        BOOST_TEST_REQUIRE(expectedSnapshotId == snapshotId);
+        BOOST_TEST_REQUIRE(expectedSnapshotDescription == snapshotDescription);
+    }
+    {
+        const std::string invalidPersistControlMessage{
+            "junk_timestamp snapshot_id invalid snapshot control message"};
+
+        ml::core_t::TTime snapshotTimestamp;
+        std::string snapshotId;
+        std::string snapshotDescription;
+        BOOST_TEST_REQUIRE(ml::api::CAnomalyJob::parsePersistControlMessage(
+                               invalidPersistControlMessage, snapshotTimestamp,
+                               snapshotId, snapshotDescription) == false);
+    }
+    {
+        const std::string invalidPersistControlMessage{" "};
+
+        ml::core_t::TTime snapshotTimestamp;
+        std::string snapshotId;
+        std::string snapshotDescription;
+        BOOST_TEST_REQUIRE(ml::api::CAnomalyJob::parsePersistControlMessage(
+            invalidPersistControlMessage, snapshotTimestamp,
+            snapshotId, snapshotDescription) == false);
+    }
+    {
+        const std::string invalidPersistControlMessage;
+
+        ml::core_t::TTime snapshotTimestamp;
+        std::string snapshotId;
+        std::string snapshotDescription;
+        BOOST_TEST_REQUIRE(ml::api::CAnomalyJob::parsePersistControlMessage(
+            invalidPersistControlMessage, snapshotTimestamp,
+            snapshotId, snapshotDescription) == false);
+    }
+    {
+        const ml::core_t::TTime expectedSnapshotTimestamp{1283524206};
+        const std::string expectedSnapshotId{"my_special_snapshot"};
+        const std::string expectedSnapshotDescription;
+
+        std::ostringstream ostrm;
+        ostrm << expectedSnapshotTimestamp << " " << expectedSnapshotId << " "
+              << expectedSnapshotDescription;
+
+        const std::string& invalidPersistControlMessage{ostrm.str()};
+
+        ml::core_t::TTime snapshotTimestamp;
+        std::string snapshotId;
+        std::string snapshotDescription;
+        BOOST_TEST_REQUIRE(ml::api::CAnomalyJob::parsePersistControlMessage(
+                               invalidPersistControlMessage, snapshotTimestamp,
+                               snapshotId, snapshotDescription) == false);
```

Comment:
This one _should_ be valid, because nothing stops a user setting an empty description.